packages/applets/calculator/src/MathInterpreter/MathInterpreter.ts

```diff
@@ -0,0 +1,224 @@
+import {createHighlightTokens, HighlightParser, highlightTags} from "@launchmenu/core";
+import {Lexer} from "@launchmenu/core/build/textFields/syntax/HighlightParser";
+import {factorial} from "./MathHelpers";
+
+const {tokens, tokenList} = createHighlightTokens({
+    lBracket: {
+        pattern: /\(/,
+        tags: [highlightTags.bracket, highlightTags.left],
+    },
+    rBracket: {
+        pattern: /\)/,
+        tags: [highlightTags.bracket, highlightTags.right],
+    },
+    add: {pattern: /\+/, tags: [highlightTags.operator]},
+    sub: {pattern: /\-/, tags: [highlightTags.operator]},
+    mul: {pattern: /\*/, tags: [highlightTags.operator]},
+    div: {pattern: /\//, tags: [highlightTags.operator]},
+    pow: {pattern: /\^|\*\*/, tags: [highlightTags.operator]},
+    percent: {pattern: /\%/, tags: [highlightTags.operator]},
+    equals: {
+        pattern: /\=/,
+        tags: [highlightTags.operator],
+    },
+    factorial: {pattern: /\!/, tags: [highlightTags.operator]},
+    paramSeperator: {pattern: /\,/, tags: [highlightTags.operator]},
+    functionName: {pattern: /[a-zA-Z_]\w+/, tags: [highlightTags.variable]},
+    value: {
+        pattern: /\-?(\d*\.)?\d+/,
+        tags: [highlightTags.literal, highlightTags.number],
+    },
+    whiteSpace: {
+        pattern: /\s+/,
+        tags: [highlightTags.whiteSpace],
+        group: Lexer.SKIPPED,
+    },
+});
+
+//assign longer_alt
+tokens.mul.LONGER_ALT = tokens.pow;
+tokens.sub.LONGER_ALT = tokens.value;
+
+export default class MathInterpreter extends HighlightParser<number> {
+    private flags: {approx: boolean};
+    constructor() {
+        super(tokenList);
+        this.performSelfAnalysis();
+        this.flags = {approx: false};
+    }
+    public variables = {};
+    public functions = {
```

Comment:
Does this class need a bit of cleaning up? Seems like this is mainly used for tokenising rather than evaluation? It might not be used at all even...